### PR TITLE
[GreenCity] - Create tests for EventsController #4736

### LINKED ISSUE #4736: [GreenCity] - Create tests for EventsController

### PR DESCRIPTION
**Pull request by sergoliarnik**
Add tests for getEventsCreatedByUser, save, update, delete, getEvent methods
Add 'jackson-datatype-jsr310' dependency to be able convert ZonedDateTime

### CODE REVIEW

**Review by NKorzhik — APPROVED** [review]

**Review by Maksym-Lenets — APPROVED** [review]

File: core/src/test/java/greencity/controller/EventsControllerTest.java [hunk 1]
```diff
@@ -16,14 +19,20 @@
...
 import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.multipart.MultipartFile;
 
 import java.security.Principal;
 
 import static greencity.ModelUtils.getPrincipal;
+import static org.mockito.Mockito.times;
 import static org.mockito.Mockito.verify;
-import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
```
Comment (RomaMocherniuk) [inline core/src/test/java/greencity/controller/EventsControllerTest.java]: please do not use `*`

File: core/src/test/java/greencity/controller/EventsControllerTest.java [hunk 2]
```diff
@@ -67,4 +76,163 @@ void getEventsCreatedByUser() {
...
+
+        MockMultipartFile jsonFile =
+            new MockMultipartFile("addEventDtoRequest", "", "application/json", json.getBytes());
+
+        mockMvc.perform(multipart(EVENTS_CONTROLLER_LINK + "/create")
+            .file(jsonFile)
+            .principal(principal)
+            .accept(MediaType.APPLICATION_JSON)
+            .contentType(MediaType.APPLICATION_JSON))
+            .andExpect(status().isCreated());
```
Comment (RomaMocherniuk) [inline core/src/test/java/greencity/controller/EventsControllerTest.java:102]: I see another `401` response code for `create` endpoint.

```
        @ApiResponse(code = 401, message = HttpStatuses.UNAUTHORIZED)
```
        
Please add a test for that cases too

File: core/src/test/java/greencity/controller/EventsControllerTest.java [hunk 3]
```diff
@@ -67,4 +76,163 @@ void getEventsCreatedByUser() {
...
+            .principal(principal))
+            .andExpect(status().isBadRequest());
+    }
+
+    @Test
+    @SneakyThrows
+    void updateTest() {
+        ObjectMapper objectMapper = new ObjectMapper();
+        objectMapper.findAndRegisterModules();
+        String json = "{\n" +
```
Comment (RomaMocherniuk) [inline core/src/test/java/greencity/controller/EventsControllerTest.java]: duplication

File: core/src/test/java/greencity/controller/EventsControllerTest.java [hunk 4]
```diff
@@ -67,4 +76,163 @@ void getEventsCreatedByUser() {
...
+            .andExpect(status().isBadRequest());
+    }
+
+    @Test
+    @SneakyThrows
+    void getEventTest() {
+        mockMvc.perform(get(EVENTS_CONTROLLER_LINK + "/event/{eventId}", 1L).principal(principal))
+            .andExpect(status().isOk());
+
+        verify(eventService)
```
Comment (RomaMocherniuk) [inline core/src/test/java/greencity/controller/EventsControllerTest.java:198]: please mock `eventService` and check the response

**Review by RomaMocherniuk — APPROVED** [review]

**Review by JJJazl — APPROVED** [review]

**Review by JJJazl — APPROVED** [review]

**Pull request merged**